asset_dashboard/static/js/components/maps/SelectAssetsMap.js

```diff
@@ -70,15 +70,33 @@ function SelectAssetsMap(props) {
   }
 
   function saveGeometries() {
-    const data = geomsToSave['features'].map(feature => {
-      return {
-        'asset_id': feature['properties']['identifier'],
-        'asset_type': searchAssetType,
-        'asset_name': feature['properties']['name'],
-        'geom': feature['geometry'],
-        'phase': phaseId
+    let existingIDs = []
+    existingGeoms.features.forEach(geom => {
+      existingIDs.push(geom.properties.asset_id)
+    })
+
+    let data = geomsToSave['features'].map(feature => {
+      // Only include feature if it doesn't already exist
+      if (!existingIDs.includes(feature.properties.identifier.toString())) {
+        return {
+          'asset_id': feature['properties']['identifier'],
+          'asset_type': searchAssetType,
+          'asset_name': feature['properties']['name'],
+          'geom': feature['geometry'],
+          'phase': phaseId
+        }
       }
-    }) 
+    })
+
+    // map() adds 'undefined' if the conditional fails, so filter those out
+    data = data.filter((item) => {
+      return item !== undefined
+    })
```

Comment:
nice filter. you can chain it to the end of your map instead of resetting the `data` value. something like
```
    let data = geomsToSave['features'].map(feature => {
      // Only include feature if it doesn't already exist
      if (!existingIDs.includes(feature.properties.identifier.toString())) {
        return {
          'asset_id': feature['properties']['identifier'],
          'asset_type': searchAssetType,
          'asset_name': feature['properties']['name'],
          'geom': feature['geometry'],
          'phase': phaseId
        }
      }
    }) 
    }).filter((item) => {
      return item !== undefined
    })
```

Reply:
ah cool, this helps clean things up, ty!